File: javascript/ql/src/experimental/Security/CWE-1427/examples/user-prompt-injection.js
const express = require("express");
const OpenAI = require("openai");

const app = express();
const client = new OpenAI();

app.get("/chat", async (req, res) => {
    let topic = req.query.topic;

    // BAD: user input is used directly in a user-role prompt
    const response = await client.chat.completions.create({
        model: "gpt-4.1",
        messages: [
            {
                role: "system",
                content: "You are a helpful assistant that summarizes topics.",
            },
            {
                role: "user",
                content: "Summarize the following topic: " + topic,
            },
        ],
    });

    res.json(response);
});

File: python/ql/src/Security/CWE-1427/examples/user-prompt-injection.py
from flask import Flask, request
from openai import OpenAI

app = Flask(__name__)
client = OpenAI()


@app.get("/chat")
def chat():
    topic = request.args.get("topic")

    # BAD: user input is used directly in a user-role prompt
    response = client.chat.completions.create(
        model="gpt-4.1",
        messages=[
            {
                "role": "system",
                "content": "You are a helpful assistant that summarizes topics.",
            },
            {
                "role": "user",
                "content": "Summarize the following topic: " + topic,
            },
        ],
    )

    return response
